profiling/src/profile/pprof/sliced_proto.rs

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+//! This file is a companion to the standard pprof protobuf definition.
+//! `Profile` fields are "sliced" into separate messages, allowing them to be
+//! serialized in a streamed manner.  This takes advantage of the fact that
+//! the top level message in a protobuf doesn't have a length header, so the
+//! bits on the wire are indistinguishable between serializing these sliced
+//! messages, and serializing the top-level message.
+//!
+//! The `tag` number and type for each of these sliced messages matches the
+//! corresponding field in the `Profile` message in `profile.proto`.
+//!
+//! Note that although many of these fields are of type `repeated` in the
+//! underlying `pprof::Profile`, there is, (except for packed arrays of scalars,
+//! which we don't use at this level), no difference in the byte representation
+//! between
+//! 1. repeatedly emitting a sliced message with a "required" field,
+//! 2. repeatedly emitting a sliced message using a "repeated" field,
+//! 3. Emitting once the message with the repeated field containing all values.
+//! In other words, we get the same bytes from "required" as "repeated", but
+//! with fewer allocations (since we don't need a `Vec` for the elements).
+//!
+//! Note that it is important to make the field "required".  Pprof requires that
+//! the "" string be the 0th element of the string table.  In a "repeated" field
+//! default-valued messages are emitted; in an "optional" field, default-valued
+//! messages are simply never emitted, and the parser just uses the default
+//! value.  This drops the "" string, and messes up the string table.
+//! Making all fields required stops this from happening and gives the desired
+//! semantics.
+
+use super::*;
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileSampleTypesEntry {
+    #[prost(message, required, tag = "1")]
+    pub sample_types_entry: ValueType,
+}
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileSamplesEntry {
+    #[prost(message, required, tag = "2")]
+    pub samples_entry: Sample,
+}
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileMappingsEntry {
+    #[prost(message, required, tag = "3")]
+    pub mappings_entry: Mapping,
+}
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileLocationsEntry {
+    #[prost(message, required, tag = "4")]
+    pub locations_entry: Location,
+}
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileFunctionsEntry {
+    #[prost(message, required, tag = "5")]
+    pub function_entry: Function,
+}
```

Comment:
Minor: For consistency with the above, I think this should be `functions_entry`

```suggestion
#[derive(Eq, Hash, PartialEq, ::prost::Message)]
pub struct ProfileFunctionsEntry {
    #[prost(message, required, tag = "5")]
    pub functions_entry: Function,
}
```

Reply:
done